internal/interaction/transaction.go

```diff
@@ -2,66 +2,181 @@ package interaction
 
 import (
 	"context"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"time"
 
-	"github.com/eurofurence/reg-payment-service/internal/domain"
-	"github.com/eurofurence/reg-payment-service/internal/repository/entities"
+	"github.com/eurofurence/reg-payment-service/internal/apierrors"
+	"github.com/eurofurence/reg-payment-service/internal/config"
+	"github.com/eurofurence/reg-payment-service/internal/entities"
 )
 
-func (s *serviceInteractor) GetTransactionsForDebitor(ctx context.Context, debitorID int64) ([]domain.Transaction, error) {
-	eTran, err := s.store.GetTransactionsByFilter(ctx, debitorID)
-	if err != nil {
-		return nil, err
+func (s *serviceInteractor) GetTransactionsForDebitor(ctx context.Context, query entities.TransactionQuery) ([]entities.Transaction, error) {
+	return s.store.GetTransactionsByFilter(ctx, query)
+}
+
+func (s *serviceInteractor) CreateTransaction(ctx context.Context, tran *entities.Transaction) (*entities.Transaction, error) {
+	mgr := NewIdentityManager(ctx)
+	if mgr.IsAdmin() || mgr.IsAPITokenCall() {
+		if tran.TransactionID == "" {
+			id, err := generateTransactionID(tran)
+			if err != nil {
+				return nil, err
+			}
+
+			tran.TransactionID = id
+		}
+
+		err := s.store.CreateTransaction(ctx, *tran)
```

Comment:
// if admin:
//    if dues:
//        -> 403 only api token may create dues

// if new payment in tentative/pending status:
//    if any tentative or pending payments already present 
//        -> 409, error message should say to delete existing tentative/pending payments first